### DISCUSSION TITLE
Got my Genmon online~

### DISCUSSION THREAD

**Discussion by Jkh62** [jgyates/genmon#1047] [Show and tell]
Sucks working in the cold in small places but having my pi hardware loaded and ready to go motivated me to go outside and hook it up.  Mid 30's today with a light wind.

The only pita part was finding a good spot to mount the Harbor Freight Pelican style case for it. I wanted it out of direct sun so between the house and the back of the generator. Pulled the panels to find an open spot and popped a hole in the back with a hole saw.  No real issues putting the hole in.

Put several lines of silicone sealer between the box and the nipple then tightened it up. Plugged in the pi, put the Gen covers back on & put my tools away. Went in the house  , washed my hands , poured a cup of coffee then sat at my PC and checked my WIFI connected devices. I gave the pi a static IP address and named it "Genmon". I expected to see it but nothing.

At first I thought it might be out of range but I was about 10' from  where I loaded it.

Turns out to power it from the generator controller a dip switch on the HAT (near the external power input on the HAT board) needed to be in the  "on" position.  Mine was set to "off" since I took a pic before closing the box. I totally missed it but figured it out.

This dip switch can be useful to switch the power off to the pi or power cycle it Vs an external switch I was considering adding.
 
Not wanting to disturb the silicon I just applied I was going to wait overnight but said screw it and waited an hour to carefully open the box and switch it on. 

When I moved the dip switch to "ON"  I saw the pi power up and got an email notification it was online before making it back in the house. WAY cool!  I should have caught the HAT dip switch settings but when I loaded the pi I powered it thru micro USB. 
Also when the Molex cable is connected WITH power you should see LED lights on the HAT & pi boards , 2nd clue I missed. 

Going to wait overnight for the silicone to cure then give it a test run and throw the transfer switch to see my Gen output stats. 
Logged an event  in the event journal for the install date to keep track on the age of the SD card.

Looking forward to tomorrows run under load to see what my generator is actually outputting.

![Genmon 1](https://github.com/jgyates/genmon/assets/157125163/1439dcc5-b3e6-4585-b9ba-cb513a8b23a1)
![Genmon 3](https://github.com/jgyates/genmon/assets/157125163/6fe9b46e-6ba2-4cc3-b697-ea15611b53eb)
![Genmon 4](https://github.com/jgyates/genmon/assets/157125163/322d9ddb-1df0-450e-89c8-15eca44d2196)

**Comment by skipfire** [comment]
The HATs ship configured for powering with the battery to barrel cable, powering over the Molex does require flipping that one switch, I'm glad you identified it fairly quickly and that the rest went well.

**Reply by Jkh62** [reply]
It really was a breeze when there is excellent documentation and help when needed.
Excellent hardware (HAT & cable). The only item I found myself looking for was small mounting screws for my pi to panel. Now I am not worried about any firmware updates to potentially kill genmon. 

The best part about the entire project :
![Screenshot_20240206_153441](https://github.com/jgyates/genmon/assets/157125163/e89690aa-6618-44d8-846b-0097f4510bb3)

**Reply by jgyates** [reply]
I am glad it is working for you. Thanks for the info and pics!

**Comment by Jkh62** [comment]
I cant put in words how pleased I am right now.
Never could have done it without your excellent coding , some hardware and a little help.

**Comment by Jkh62** [comment]
A pic is an easier reminder if you chose not to read instructions , like I did.

![MOLEX](https://github.com/jgyates/genmon/assets/157125163/b3c2215e-d699-44a2-9840-661378c320a0)

**Reply by skipfire** [reply]
I'm working on an update to the instructions that will include some diagrams in addition to the configuration instructions.

**Reply by Jkh62** [reply]
The instructions are excellent.
I just did not read them and was focused on loading the pi correctly and the vids I watched showed people just plugging the cable in or powering it from another source. I never gave it a second thought.

Totally my fault and the clue should have been no LED light when I plugged the Molex connectors in.

Thx for making a great HAT !

**Reply by skipfire** [reply]
The videos were made with an earlier version of the pHATs that did not include those switches, which is why I'm updating the instructions. I'm also making my own collection of instruction videos that will be 2 to 5 minutes each with each one being one section of the process.
- Physically prepping the Pi + HAT/pHAT - connecting them with the standoffs, configuring the switches, etc
- Setting up WiFi through the GUI
- Setting up WiFi through the captive portal (for pre-loaded units)
- Physically installing in the generator
- Probably one installing GenMon and running the serial tests
I've already got one made (though I haven't linked it yet) and I'm making a second video today.

**Comment by Jkh62** [comment]
Excellent !

**Comment by Jkh62** [comment]
Ran my 1st test , maintenance run ( manual)  with load transfer for 20 minutes.
Picture perfect and good to see what my system is actually outputting.

![image](https://github.com/jgyates/genmon/assets/157125163/2cd8f6ae-383b-4de3-97f0-acc6c0c52d9d)

**Reply by skipfire** [reply]
My house never has that low of power draw, glad you got it all working.

**Comment by Jkh62** [comment]
Is there a register in the generator controller for transfer switch status as in open / closed ? It would be a nice added indicator on the status screen for the transfer switch as open / closed. I was watching the power output indicator to see when it engaged and the power output to drop when it went open.  Really impressed at genmon and its operation.  Wish I knew more about the inside function and coding to be of help. Did the Log2ram mod to take some load off the SD card and after I got the log file size tweaked its running correctly.

**Comment by jgyates** [comment]
that depends on your generator model. For the air cooled Generac Evolution, no. The transfer switch status is not available over modbus.

**Comment by Jkh62** [comment]
Bummer but still awesome software and execution.

**Comment by jgyates** [comment]
The only generac consumer stand by generator that has the abilityy to read the relay status that signals the transfer switch is the Evolution Liquid Cooled models and this only reads the value written to the relay, not the actual status of the transfer switch.

**Comment by Jkh62** [comment]
The liquid cooled Gen was my original pick for a generator, until I talked to my gas company (Columbia Gas) and they wanted about $900 to put a larger gas meter on my house. My gas meter is original to my house built in 1996. In principle I can't pay for their meter to use more of their product.

After rethinking it I went with the air cooled generac 22KW model. If I needed something to run 24X7 the lower rpm liquid cooled is the better pick but I have not ever had outages that long.

Who knows what future firmware updates will open or close but VERY happy with my system. 


Elevated my router in the house about 2 ft.  and boosted my WIFI signal to solid green .
Woohoo!

![image](https://github.com/jgyates/genmon/assets/157125163/8bab9f51-bab0-4877-908f-6bc3125722c2)

**Comment by Jkh62** [comment]
Genmon is running great . I had some time to tweak my WIFI settings for maximum signal strength.
I am running a pi zero W and found my strongest signal  is setting my router on channel #1 / 2.4Ghz  band . 20 MHz only.
Average signal is -60 dBm with the above settings. 

My CPU use is around 14%.

**Comment by jgyates** [comment]
Thanks for the info. I know there have been a few folks who have had weak wifi. I am glad this helped.
